### PR TITLE
[FIRRTL][LowerToHW] Remove DontTouch annotation after lowering 

### PR DESCRIPTION
**Pull request by prithayan**
This change is required to ensure that no annotations remain unprocessed in FIRRTL dialect, after lowering to `HW`.
Remove the `DontTouch` annotation after lowering it to symbols in `LowerToHW`.
Add a function in `AnnotationSet` to remove annotations of a given class name.
There will be another PR, to check for unprocessed annotations, which should also serve as a test case for this PR.

### CODE REVIEW

File: test/Conversion/FIRRTLToHW/lower-to-hw-errors.mlir [hunk 1]
```diff
@@ -19,4 +19,4 @@ firrtl.circuit "InvalidBundle" {
   func private @UnknownFunction() {
     return
   }
-}
\ No newline at end of file
+}
```
Comment (seldridge) [inline test/Conversion/FIRRTLToHW/lower-to-hw-errors.mlir]: Nit: unrelated whitespace change.

**Review by seldridge — APPROVED** [review]

**Comment by prithayan** [comment]
Thanks @seldridge , Merged.
